WalletWasabi.Gui/ViewModels/Validation/ReflectionHelper.cs

```diff
@@ -50,6 +50,22 @@ public static T InvokeMethod<T>(object instance, string methodName)
 			}
 		}
 
+		public static MethodInfo GetMethodInfo<T>(object instance, string methodName)
+		{
+			MethodInfo info = instance.GetType().GetRuntimeMethod(methodName, new Type[0]);
+
+			if (info != null &&
+				info.ReturnType == typeof(T) &&
+				info.GetParameters().Length == 0)
+			{
+				return info;
+			}
+			else
+			{
+				throw new ArgumentException("Method was not found on class");
+			}
+		}
+
```

Comment:
Reflection is generally to be avoided. Is this absolutely necessary?

Reply:
Yes, i think this, for a fact, reduces reflection calls because we only scan for methods via reflection on startup then the information about properties with validation attributes are cached into ViewModelBase. Instead of calling reflection to scan for properties every single validation call, we just use the cache to match property names to the MethodInfo that we saved.